src/daemon/rpmostreed-os-experimental.cxx

```diff
@@ -207,20 +208,38 @@ osexperimental_handle_download_packages (RPMOSTreeOSExperimental *interface,
     goto out;
 
   sack = dnf_context_get_sack (rpmostree_context_get_dnf (ctx));
-
-  for (const char* const* it = queries; it && *it; it++)
+  RpmOstreeOverrideSourceKind source_kind;
+  source_name = static_cast<char*> (rpmostree_parse_override_source (source, &source_kind, &local_error));
+  if (local_error)
+    goto out;
+  if (source_kind == RPM_OSTREE_OVERRIDE_SOURCE_KIND_REPO)
     {
-      auto query = static_cast<const char*> (*it);
-      g_autoptr(GPtrArray) pkglist = rpmostree_get_matching_packages (sack, query);
-
-      if (!pkglist || pkglist->len == 0)
+      for (const char* const* it = queries; it && *it; it++)
         {
-          glnx_throw (&local_error, "No matching packages for \"%s\"", query);
-          goto out;
+          g_autoptr(GPtrArray) pkglist = NULL;
+          HyNevra nevra = NULL;
+          g_auto(HySubject) subject = hy_subject_create(static_cast<const char*> (*it));
+          hy_autoquery HyQuery query =
+            hy_subject_get_best_solution(subject, sack, NULL, &nevra, FALSE, TRUE, TRUE, TRUE, FALSE);
+          hy_query_filter (query, HY_PKG_REPONAME, HY_EQ, source_name);
+          pkglist = hy_query_run (query);
+          if (!pkglist || pkglist->len == 0)
+            {
+              glnx_throw (&local_error, "No matches for \"%s\" in repo '%s'", (char*) it, source_name);
```

Comment:
I think this should be

```suggestion
              glnx_throw (&local_error, "No matches for \"%s\" in repo '%s'", (char*) *it, source_name);
```

Reply:
In general, it's nicer to deference the iterator upfront and store that in a variable, as you had before with `auto query`.